src/objects/viewport.ts

```diff
@@ -40,6 +40,16 @@ export async function startViewport(): Promise<void> {
       minWidth: stage.clientWidth / 2,
     })
 
+  // Both the `moved` and `moved-end` events are required for a reliable `viewportMoved` event call that
+  // includes the final positions for both scroll wheel and touch pad interactions.
+  viewport
+    .on('moved', () => {
+      emitter.emit('viewportMoved', null)
+    })
+    .on('moved-end', () => {
+      emitter.emit('viewportMoved', null)
+    })
```

Comment:
Something I didn't mention in the video. I did try and pass the event returned from the `moved` and `moved-end` callbacks, which include a reference to the viewport, for use in the update calls thinking maybe they return a non-delayed value. Unfortunately I got the same result as simply referencing the `viewport` object directly.